### ISSUE TITLE
Unsupported Resource Type "azurerm_automation_software_update_configuration"

### ISSUE THREAD

**Issue by Isaac-Toovey** [hashicorp/terraform-provider-azurerm#19789]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.6

### AzureRM Provider Version

3.17.0

### Affected Resource(s)/Data Source(s)

azurerm_automation_software_update_configuration

### Terraform Configuration Files

```hcl
resource "azurerm_automation_account" "automation" {
  for_each            = var.accounts
  name                = each.value.name
  location            = var.location
  resource_group_name = var.resource_group_name
  sku_name            = each.value.sku

  dynamic "identity" {
    for_each = var.identity
    content {
      type = identity.value.type
    }
  }
}

resource "azurerm_automation_runbook" "runbooks" {
  for_each                = var.runbooks
  name                    = each.value.name
  location                = var.location
  resource_group_name     = var.resource_group_name
  automation_account_name = each.value.automation_account_name
  log_verbose             = each.value.log_verbose
  log_progress            = each.value.log_progress
  description             = each.value.description
  runbook_type            = each.value.runbook_type

  content = each.value.content
}

resource "azurerm_automation_software_update_configuration" "Update_configuration" {
  for_each              = var.Update_configuration
  name                  = each.value.Update_configuration
  automation_account_id = azurerm_automation_account.automation.id
  operating_system      = each.value.operating_system
  Windows {
    classification    = "Security"
    reboot            = "Never"
  }
}
```

### Debug Output/Panic Output

```shell
PS C:\Users\XXXXX\Documents\TF\XXXXXX\terraform\Single_resource_deploy\automation-accounts> terraform plan -var-file="./environments/prod-1.tfvars"
╷
│ Error: Invalid resource type
│
│   on main.tf line 43, in resource "azurerm_automation_software_update_configuration" "automation_software_update_configuration":
│   43: resource "azurerm_automation_software_update_configuration" "automation_software_update_configuration" {
│
│ The provider hashicorp/azurerm does not support resource type "azurerm_automation_software_update_configuration".
╵
PS C:\Users\XXXXX\Documents\TF\XXXXXX\terraform\Single_resource_deploy\automation-accounts>
```

### Expected Behaviour

successful TF plan

### Actual Behaviour

Failed Plan due to unsupported Resource Type

### Steps to Reproduce

insert the relevant code block for azurerm_automation_software_update_configuration
run tf plan

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/automation_software_update_configuration#timeouts

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. This new resource is introduced to TF since v3.24.0. So suggest to upgrade azurerm provider to latest version.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of v3.24.0.
